File: Release/WetiMatlabFunctions/PreProcessing/AdjustInputFiles.m
function AdjustInputFiles(InputFiles,Modifications,SimulationFolder)

n_File                  = size(InputFiles,1);
n_Modification          = size(Modifications,1);    

for i_File = 1:n_File
    TemplateFile        = InputFiles{i_File,1};
    NewFile             = InputFiles{i_File,2};
    copyfile(fullfile(SimulationFolder,TemplateFile),fullfile(SimulationFolder,NewFile));
end

for i_Modification = 1:n_Modification
    i_File                  = str2double(Modifications{i_Modification,1});
    Mode                    = Modifications{i_Modification,2};
    NewFile                 = InputFiles{i_File,2};
    TXTFile                 = fullfile(SimulationFolder,NewFile);
    switch Mode
        case 'I' % Identifier
            Identifier      = Modifications{i_Modification,3};
            NewString       = Modifications{i_Modification,4};
            ManipulateFastInputFile(TXTFile,Identifier,NewString);
        case 'R' % Replace
            StringToReplace = Modifications{i_Modification,3};
            NewString       = Modifications{i_Modification,4};
            ManipulateTXTFile(TXTFile,StringToReplace,NewString); 
        case 'A' % Add line
            nLine           = str2double(Modifications{i_Modification,3});
            NewLine         = Modifications{i_Modification,4};            
            AddLineToTXTFile(TXTFile,nLine,NewLine)
    end
end

end

File: Release/PythonFunctions/AdjustinputFiles.py
import os
import shutil
from ManipulateFastInputFile import ManipulateFastInputFile
from ManipulateTXTFile import ManipulateTXTFile
from AddLineToTXTFile import AddLineToTXTFile
def adjust_input_files(input_files, modifications, simulation_folder):
    n_file = len(input_files)
    n_modification = len(modifications)

    for i_file in range(n_file):
        template_file = input_files[i_file][0]
        new_file = input_files[i_file][1]
        shutil.copyfile(os.path.join(simulation_folder, template_file), os.path.join(simulation_folder, new_file))

    for i_modification in range(n_modification):
        i_file = int(modifications[i_modification][0])
        mode = modifications[i_modification][1]
        new_file = input_files[i_file][1]
        txt_file = os.path.join(simulation_folder, new_file)
        if mode == 'I':  # Identifier
            identifier = modifications[i_modification][2]
            new_string = modifications[i_modification][3]
            ManipulateFastInputFile(txt_file, identifier, new_string)
        elif mode == 'R':  # Replace
            string_to_replace = modifications[i_modification][2]
            new_string = modifications[i_modification][3]
            ManipulateTXTFile(txt_file, string_to_replace, new_string)
        elif mode == 'A':  # Add line
            n_line = int(modifications[i_modification][2])
            new_line = modifications[i_modification][3]
            AddLineToTXTFile(txt_file, n_line, new_line)
